android/src/com/frostwire/android/gui/activities/MainActivity.java

```diff
@@ -139,54 +141,133 @@
     private BroadcastReceiver mainBroadcastReceiver;
     private boolean externalStoragePermissionsRequested = false;
 
-    public MainActivity() {
-        super(R.layout.activity_main);
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
         this.controller = new MainController(this);
-        this.fragmentsStack = new Stack<>();
-        this.permissionsCheckers = initPermissionsCheckers();
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_main);
+        setSupportActionBar(toolbar);
+
+        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
```

Comment:
might want to keep this and use it on the `toggleDrawer()` method to be invoked on `KeyEvent.KEYCODE_MENU`